src/magicbox/SuggestionsManager.ts

```diff
@@ -420,70 +371,13 @@ export class SuggestionsManager {
     return suggestion;
   }
 
-  private moveWithQuerySuggestPreview(direction: Direction) {
-    const currentlySelected = $$(this.element).find(`.${this.options.selectedClass}`);
-    const omniboxSelectables = $$(this.element).findAll(`.${this.options.suggestionClass}`);
-    const previewSelectables = $$(this.element).findAll('.coveo-preview-selectable');
-    const omniboxIndex = indexOf(omniboxSelectables, currentlySelected);
-    const previewIndex = indexOf(previewSelectables, currentlySelected);
-
-    const verticalMove = direction === Direction.Up || direction === Direction.Down;
-    const suggestionIsSelected = omniboxIndex > -1;
-    if (suggestionIsSelected && verticalMove) {
-      this.moveWithinSuggestion(direction);
-      return;
-    }
-
-    const noPreviewDisplayed = previewSelectables.length === 0;
-    const directionIsLeft = direction === Direction.Left;
-    const noPreviewSelected = previewIndex === -1;
-    if (noPreviewDisplayed || (directionIsLeft && noPreviewSelected)) {
-      return;
-    }
-    this.moveWithinPreview(direction);
-  }
-
   private moveWithinPreview(direction: Direction) {
-    const previewSelectables = $$(this.element).findAll('.coveo-preview-selectable');
-
-    let newSelectedIndex;
-    if (direction === Direction.Up || direction === Direction.Down) {
-      newSelectedIndex = this.moveVerticallyInPreview(direction);
-    }
-
-    if (direction === Direction.Left || direction === Direction.Right) {
-      newSelectedIndex = this.moveHorizontallyInPreview(direction);
-    }
-    if (Utils.isNullOrUndefined(newSelectedIndex)) {
+    const newFocusedPreview = this.resultPreviewsManager.getElementInDirection(direction);
+    if (!newFocusedPreview) {
+      this.selectQuerySuggest(this.resultPreviewsManager.previewsOwner);
       return;
     }
```

Comment:
`newFocusedPreview` will be null if the selection is out of bounds and shouldn't wrap (e.g., the first preview is selected and the left arrow key is pressed).